diffpy/snmf/tests/test_factorizers.py

```diff
@@ -4,9 +4,10 @@
 from diffpy.snmf.factorizers import lsqnonneg
 
 tl = [
-    ([[[1, 0], [1, 0], [0, 1]], [2, 1, 1]],[1.5, 1.]),
+    ([[[1, 0], [1, 0], [0, 1]], [2, 1, 1]], [1.5, 1.]),
     ([[[2, 3], [1, 2], [0, 0]], [7, 7, 2]], [0, 2.6923]),
-    ([[[3,2,4,1]],[3.2]],([0,0,.8,0])),
+    ([[[3, 2, 4, 1]], [3.2]], [0, 0, .8, 0]),
```

Comment:
This is great, I can merge like this, but the general rule is have one test for each thing you want to test, so when I mentioned about testing someone passing in lists, we only needed one test that looked like that.   It is also possible to test bad inputs, but it takes more effort and there is not enough to test here because we are just passing the exception on up from numpy so there is not so much value in doing that.  These are just general comments.  This is the test-driven approach mindset:
1) Think about what behavior your want your function to have
2) write a test that tests that behavior
3) think of any edge cases you want to test for (e.g., unexpected things being passed in).  Only test them if you are "Handling" them.  For example, if you have a statement like `if passed_in_variable is None: <set it to somehting>`.  IF you handle them by rasing a particular exception you cna test that too, but it is rare.
4) write the function until all the tests pass and you are done.

Can you maybe start thinking in this way?  You are nearly there.

Reply:
I understand. I will start writing more robust tests and include edge cases.